### PR TITLE
Data apps layout updates

### PR DESCRIPTION
**Pull request by kdoh**
Some fairly substantial updates to data apps UX.

1. App navigation now appears in a navbar across the top. This makes it significantly easier to browse through the app.
2. You can access the "contents" of your app, including models by clicking on a trigger at the top right of the nav.
3. App pages have a max-width, to make layout choices easier to reason about.

### CODE REVIEW

File: frontend/src/metabase/dashboard/components/DashboardGrid.jsx [hunk 1]
```diff
@@ -68,7 +68,7 @@ class DashboardGrid extends Component {
   };
 
   static defaultProps = {
-    width: 0,
+    width: 1200,
```
Comment (kdoh) [inline frontend/src/metabase/dashboard/components/DashboardGrid.jsx:71]: Need to set this based on `is_app_page`

File: frontend/src/metabase/dashboard/components/DashboardGrid.jsx [hunk 2]
```diff
@@ -366,8 +366,8 @@ class DashboardGrid extends Component {
         layouts={layouts}
         breakpoints={GRID_BREAKPOINTS}
         cols={GRID_COLUMNS}
-        width={width}
-        margin={{ desktop: [6, 6], mobile: [6, 10] }}
+        width={1200}
```
Comment (kdoh) [inline frontend/src/metabase/dashboard/components/DashboardGrid.jsx:369]: Ditto this.

File: frontend/src/metabase/dashboard/components/DashboardGrid.jsx [hunk 3]
```diff
@@ -366,8 +366,8 @@ class DashboardGrid extends Component {
...
+        margin={{ desktop: [32, 18], mobile: [6, 10] }}
```
Comment (kdoh) [inline frontend/src/metabase/dashboard/components/DashboardGrid.jsx:370]: Need to make this data app contextual.

File: frontend/src/metabase/dashboard/components/DashboardHeader.tsx [hunk 4]
```diff
@@ -136,6 +136,12 @@ const DashboardHeader = ({
         isNavBarOpen={isNavBarOpen}
         className={cx("QueryBuilder-section", headerClassName)}
         ref={header}
+        style={{
```
Comment (kdoh) [inline frontend/src/metabase/dashboard/components/DashboardHeader.tsx:139]: Probably best to move this into the component itself.

File: frontend/src/metabase/lib/dashboard_grid.js [hunk 5]
```diff
@@ -1,5 +1,5 @@
 export const GRID_WIDTH = 18;
-export const GRID_ASPECT_RATIO = 4 / 3;
+export const GRID_ASPECT_RATIO = 21 / 9;
```
Comment (kdoh) [inline frontend/src/metabase/lib/dashboard_grid.js:2]: Should set this for data apps only.

File: frontend/src/metabase/lib/dashboard_grid.js [hunk 6]
```diff
@@ -15,7 +15,7 @@ export const GRID_COLUMNS = {
 
 export const DEFAULT_CARD_SIZE = { width: 4, height: 4 };
 
-export const MIN_ROW_HEIGHT = 54;
+export const MIN_ROW_HEIGHT = 42;
```
Comment (kdoh) [inline frontend/src/metabase/lib/dashboard_grid.js:18]: Ditto this

File: frontend/src/metabase/visualizations/components/List/List.styled.tsx [hunk 7]
```diff
@@ -19,6 +19,11 @@ export const Root = styled.div<{ isQueryBuilder?: boolean }>`
     css`
       margin: 2rem 6rem;
     `}
+  td {
```
Comment (kdoh) [inline frontend/src/metabase/visualizations/components/List/List.styled.tsx:22]: I think this declaration is causing some issues, might be best to remove it for now

File: frontend/src/metabase/dashboard/containers/DashboardApp.jsx [hunk 8]
```diff
@@ -162,6 +169,13 @@ const DashboardApp = props => {
 
   return (
     <DataAppContext>
+      {dashboard && dashboard.is_app_page && (
```
Comment (kdoh) [inline frontend/src/metabase/dashboard/containers/DashboardApp.jsx:172]: Need to be able to show / hide this based on `isEditing`

File: frontend/src/metabase/dashboard/components/DashboardGrid.jsx [hunk 9]
```diff
@@ -366,8 +377,12 @@ class DashboardGrid extends Component {
         layouts={layouts}
         breakpoints={GRID_BREAKPOINTS}
         cols={GRID_COLUMNS}
-        width={width}
-        margin={{ desktop: [6, 6], mobile: [6, 10] }}
+        width={dashboard.is_app_page ? DATA_APPS_MAX_WIDTH : width}
```
Comment (kdoh) [inline frontend/src/metabase/dashboard/components/DashboardGrid.jsx:380]: At some point I feel like we're going to want to just fork the `DashboardGrid` component. I think that'd make it easier to reason about.

File: frontend/src/metabase/dashboard/components/Dashboard/Dashboard.jsx [hunk 10]
```diff
@@ -260,29 +266,41 @@ class Dashboard extends Component {
...
-                {shouldRenderParametersWidgetInEditMode && (
-                  <ParametersWidgetContainer
-                    data-testid="edit-dashboard-parameters-widget-container"
-                    isEditing={isEditing}
-                  >
-                    {parametersWidget}
-                  </ParametersWidgetContainer>
+              <>
+                {isDataApp && !isEditing && (
+                  <div className="bg-white border-bottom py1">
```
Comment (kdoh) [inline frontend/src/metabase/dashboard/components/Dashboard/Dashboard.jsx:271]: Scoot over to emotion.

File: frontend/src/metabase/dashboard/containers/DashboardHeader.jsx [hunk 11]
```diff
@@ -432,7 +432,32 @@ class DashboardHeader extends Component {
...
+                  key="edit-dashboard"
+                  tooltip={
+                    dashboard.is_app_page ? t`Edit page` : t`Edit dashboard`
+                  }
+                >
+                  <DashboardHeaderButton
+                    key="edit"
+                    data-metabase-event="Dashboard;Edit"
+                    icon="pencil"
+                    className="text-white cursor-pointer bg-brand p2 m2"
```
Comment (kdoh) [inline frontend/src/metabase/dashboard/containers/DashboardHeader.jsx:448]: Scoot over to emotion

File: frontend/src/metabase/nav/containers/MainNavbar/DataAppNavbar/DataAppNavbarView.tsx [hunk 12]
```diff
@@ -74,23 +73,25 @@ function DataAppNavbarView({
...
-        </ul>
-      </SidebarSection>
-      <DataAppActionPanel
-        dataApp={dataApp}
-        onAddData={onAddData}
-        onNewPage={onNewPage}
-        onEditAppSettings={onEditAppSettings}
-      />
-    </SidebarContentRoot>
+    <div className="flex align-center">
```
Comment (kdoh) [inline frontend/src/metabase/nav/containers/MainNavbar/DataAppNavbar/DataAppNavbarView.tsx:76]: Scoot over to emotion.

**Review by kulyk — APPROVED** [review]

**Pull request merged**